### ISSUE TITLE
[False positive]  Property Class::$request (Illuminate\Http\Request) does not accept Illuminate\Support\Facades\Request.

### ISSUE THREAD

**Issue by hmingv** [nunomaduro/larastan#1264]
- Larastan Version: 2.1.6
- `--level` used: 6
- Pull request with failing test: null

### Description

`\Illuminate\Support\Facades\Request::instance()` method returns `\Illuminate\Http\Request` instance, but Larastan detects `\Illuminate\Support\Facades\Request`

### Laravel code where the issue was found

```php
class Upload {
    /**
     * Request
     *
     * @var \Illuminate\Http\Request
     */
    protected \Illuminate\Http\Request $request;

    public function __construct()
    {
        $this->request = \Illuminate\Support\Facades\Request::instance();
    }
}
```

### Analysis results
```
 ------ --------------------------------------------------------------------------------------------------------------------- 
  Line   Library\Upload.php                                                                                                   
 ------ ---------------------------------------------------------------------------------------------------------------------
  97     Property App\Library\Upload::$request (Illuminate\Http\Request) does not accept Illuminate\Support\Facades\Request.
 ------ ---------------------------------------------------------------------------------------------------------------------
```

**Comment by hmingv** [comment]
I found a similar (ISSUE)[https://github.com/nunomaduro/larastan/issues/251] but not sure how to fix this.

**Comment by hmingv** [comment]
I found a similar [ISSUE](https://github.com/https://github.com/nunomaduro/larastan/issues/251) but not sure how to fix this.

**Comment by szepeviktor** [comment]
Hello @hmingv!

Isn't automatic injection working? `public function __construct(Request $request)` ?

**Comment by hmingv** [comment]
Thank you @szepeviktor for your reply.

Automatic injection is in effect, this Upload instance was created manually by me, so it is not used (if I use automatic injection instead, I don't think there will be this problem).

But I want to see if this can be fixed on larastan, I think it analyse wrong.

**Comment by szepeviktor** [comment]
I do not no the answer.

Seeing Laravel code I infer that your code should use `request()` global or auto-injection.
Larastan does not *yet* cover the whole framework.

**Comment by canvural** [comment]
Fixed with 55b228f3401048adab2f61a90aaedd09cffe7923
